eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java

```diff
@@ -139,6 +145,60 @@ private void getOrCreateClusteredVariantAccessions(List<? extends SubmittedVaria
         checkForMerges(submittedVariantEntities);
     }
 
+    /**
+     * This method handles a specific scenario in clustering where an already clustered variant is remapped and
+     * no matching rs id exist. In this case, clustered variant should continue using its existing rs id and an entry
+     * needs to be made ClusteredVariantEntity collection.
+     *
+     *                     SubmittedVariantEntity                                                           ClusteredVariantEntity
+     * SS	HASH	        RS	ASM	    STUDY	CONTIG	POS	REF	ALT                             RS	HASH	        ASM	    POS	CONTIG	TYPE
+     * 500	ASM1_Chr1_1_A_T	300	ASM1	PRJEB1	Chr1	1	A	T    (original)                 300	ASM1_Chr1_1_SNV	ASM1	1	Chr1	SNV
+     * 500	ASM2_Chr1_2_A_T	300	ASM2	PRJEB1	Chr1	2	A	T    (remapped to asm2)
+     *
+     * No RS id exist for the HASH ASM2_Chr1_2_A_T,	but it has an existing rs id of 300, which it should continue to use.
+     * It should not generate a new rs id for this situation as they will get merged right after generation and will be wasted
+     *
+     *         ClusteredVariantEntity (after operation)
+     *          RS	HASH	        ASM	    POS	CONTIG	TYPE
+     *          300	ASM1_Chr1_1_SNV	ASM1	1	Chr1	SNV
+     *          300	ASM2_Chr1_2_SNV	ASM1	1	Chr1	SNV
+     */
+    private List<IClusteredVariant> processClusteredVariantsWhereNoRSExists(List<? extends SubmittedVariantEntity> submittedVariantEntities) {
+        List<SubmittedVariantEntity> submittedVariantWithRS = submittedVariantEntities.stream()
```

Comment:
I am not sure about this part. We separated the mongo readers to process first clustered and then unclustered variants so the filter is already happening at the database level but we are checking it again here.

Reply:
Again the writer is same, so the same method will get called in case of both the steps. This method does not make sense in case of step 2 as all variants will be non clustered so should get filtered out. 
we can take the same approach as reader and create 2 different beans for writers with one Boolean field, that will determine whether we need to call this method or not.

Reply:
I would rather that instead and not filtering the variants again. That way is clear in which cases it should not be called

Reply:
updated